messaging/usecase/room_usecase.go

```diff
@@ -4,61 +4,105 @@ import (
 	"chat/domain"
 	"chat/utils/errors"
 	"fmt"
-	"log"
 	_ "reflect"
 	"time"
 )
 /*
 
 	SaveRoom( room *ChatRoom) error
+	AddUserToRoom(roomID string, studentID string) error
+	RemoveUserFromRoom(roomID string, studentID string) error
 	GetChatRoomsFor( studentID string) (*StudentChatRooms, error)
-	EditChatRoomParticipants( room *ChatRoom) error
 	DeleteRoom( userID string, roomID string) error
 */
 type roomUseCase struct {
 	rr      domain.RoomRepository
-	st 		domain.StudentRepository
+	sr 		domain.StudentRepository
 	timeout time.Duration
 }
 
+func NewRoomUseCase(rr domain.RoomRepository, sr domain.StudentRepository, t time.Duration) domain.RoomUseCase {
+	return &roomUseCase{rr: rr, sr: sr, timeout: t}
+}
+
+// SaveRoom should add room to chat.room & chat.student_rooms for all participants
 func (u *roomUseCase) SaveRoom(room *domain.ChatRoom) error {
 
 	_,err := u.rr.GetRoom(room.RoomID)
 	if err==nil {
 		return errors.NewConflictError(fmt.Sprintf("Room with ID %s already exists", room.RoomID))
 	}
-	_,err=u.st.GetStudent(room.Admin.ID)
-	if err==nil {
+	_,err=u.sr.GetStudent(room.Admin.ID)
+	if err!=nil {
 		return errors.NewConflictError(fmt.Sprintf("The admin with ID  %s does not exist", room.Admin.ID))
 	}
 	err = u.rr.SaveRoom( room )
 	if err!=nil {
 		return err
 	}
 
-   return err
+	var studentIDs []string
+	for _,student := range room.Students {
```

Comment:
If we check that the admin exists, do we also check if every student exists?

Reply:
great point, I'll check if each participant exists instead of just admin

Reply:
Can you look [here](func (i *inventoryUseCase) fillItemDetails(c context.Context, inventoryItems []domain.InventoryItem) ([]domain.InventoryItem, error) {) for suggestions on how to do it concurrently using goroutines. For-loop will be really slow.

Reply:
Actually i'm not sure about really slow. We will be adding a max of what 5-6 people at a time? won't make a difference i think. YAGNI maybe